Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/FareCalculator.hs

```diff
@@ -252,19 +258,23 @@ calculateFareParameters params = do
     countPlatformFee :: Money -> Maybe PlatformFeeInfo -> FareParametersDetails -> FareParametersDetails
     countPlatformFee fullCompleteRideCost platformFeeInfo = \case
       (DFParams.ProgressiveDetails det) -> DFParams.ProgressiveDetails det -- should be impossible anyway
-      (DFParams.SlabDetails det) ->
-        DFParams.SlabDetails det {platformFee = countPlatformFeeMath <$> platformFeeInfo}
+      (DFParams.SlabDetails _det) ->
+        DFParams.SlabDetails $ maybe (FParamsSlabDetails Nothing Nothing Nothing) countPlatformFeeMath platformFeeInfo
       where
         countPlatformFeeMath platformFeeInfo' = do
           let baseFee = case platformFeeInfo'.platformFeeCharge of
                 ProgressivePlatformFee charge -> fromIntegral fullCompleteRideCost * realToFrac charge
                 ConstantPlatformFee charge -> fromIntegral charge
-          roundToIntegral (baseFee + baseFee * platformFeeInfo'.cgst + baseFee * platformFeeInfo'.sgst)
+          FParamsSlabDetails
+            { platformFee = Just . roundToIntegral $ baseFee, -- when we round each component separately, rounding errors will increase
```

Comment:
we can keep them as double itself and round at then end when requried.

Reply:
`sgst` and `cgst` changed to `HighPrecMoney` as you suggested